File: 2748-number-of-beautiful-pairs/2748-number-of-beautiful-pairs.py
class Solution:
    def countBeautifulPairs(self, nums: List[int]) -> int:
        ans = 0
        lst = []
        fst = []
        for i in range (len(nums)):
            lst.append(nums[i]%10);
            while nums[i] > 9:
                nums[i]//=10
        for i in range (len(nums)):
            for j in range(i+1,len(nums)):
                if  gcd(nums[i],lst[j]) == 1:
                    ans +=1
        return ans

File: 2748-number-of-beautiful-pairs/2748-number-of-beautiful-pairs.cpp
class Solution {
public:
    int countBeautifulPairs(vector<int>& nums) {
        vector<int>fst,lst;
        for(int x:nums){
            fst.push_back(x%10);
            while(x>9)
                x/=10;
            lst.push_back(x);
        }
        int ans =0;
        for(int i=0;i<nums.size();++i){
            for(int j=i+1;j<nums.size();++j){
                if(__gcd(lst[i],fst[j])==1)
                    ++ans;
            }
        }
        return ans;
    }
};